warn/warn_macro.go

```diff
@@ -0,0 +1,334 @@
+// Warnings for using deprecated functions
+
+package warn
+
+import (
+	"fmt"
+
+	"github.com/bazelbuild/buildtools/build"
+)
+
+// Internal constant that represents the native module
+const nativeModule = "<native>"
+
+// acceptsNameArgument checks whether a function can accept a named argument called "name",
+// either directly or via **kwargs.
+func acceptsNameArgument(def *build.DefStmt) bool {
+	for _, param := range def.Params {
+		switch param := param.(type) {
+		case *build.Ident:
+			if param.Name == "name" {
+				return true
+			}
+		case *build.AssignExpr:
+			if ident, ok := param.LHS.(*build.Ident); ok && ident.Name == "name" {
+				return true
+			}
+		case *build.UnaryExpr:
+			if param.Op == "**" {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// fileData represents information about rules and functions extracted from a file
+type fileData struct {
+	rules     map[string]int
+	functions map[string]map[string]funCall
+	aliases   map[string]alias
```

Comment:
Is it useful to have a separate field for aliases?

I could imagine that aliases would show up as rules/functions.

Reply:
Yes, because an alias can refer to a loaded function (if it's been re-exported).